### PR TITLE
control_of_documents

### PR DESCRIPTION
**Pull request by RxLoutre**
Here is my progress on my part ! I have not written one paragraph, because I would first like some feedback. I want to use images, and I also don't know if I am going the right direction. I am not sure what is expected. So I am waiting to be able to discuss with the group to write more :)

### CODE REVIEW

**Comment by Bratdaking** [comment]
Sorry late reaction because I only saw this request today, but there are handy cheatsheets for markdown (md):
for example: https://www.markdownguide.org/cheat-sheet

This is how you can include a figure: alt text
Put the figure within the same folder and it will be included.

File: docs/fieldnorm/control_of_documents.md [hunk 1]
```diff
@@ -1,25 +1,65 @@
...
+9. One paper or electronic copy of each obsolete controlled document is retained for a specified time period or in accordance with applicable specified requirements
+
+## Why do Bioinformaticians need a different QMS
+
+Bioinformaticians create workflows collaboratively on a computer within a git environment. Git is a distributed version control system that tracks changes in any set of computer files. This can be used to track the changes of the code itself, but also in its documentation, which can describe the Standard Operating Procedures to use it. Git comes in with a set of features allowing to track issues and their resolution, new release of a workflow and its newest features. As the workflow often get upgraded and updated, their documentation, as well, is susceptible to change. Which is why many Bioinforaticians have turned their back from a standard QMS to have their own documentation system, often linked to git. Such system also offer possibilities to copy-paste code blocks while performing the SOP for a given flow, making it so much easier and less prone to errors. Examples of those documentations systems are:
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: change "turned their back" to something positive. I.e.. "Which is why many Bioinformaticians choose to use their own documentation system, which in line with the development of code"

Comment (hwmvandeutekom) [reply docs/fieldnorm/control_of_documents.md]: I would make this the start of the document; to immediately make the point that we deviate from what is used in the wetlab, and why we do this.

File: docs/fieldnorm/control_of_documents.md [hunk 2]
```diff
@@ -1,25 +1,65 @@
...
-- Software development, and hence, corresponding documentation is prone to frequent updates. With the QMS of the laboratory, there are often waiting steps for different members of the laboratory to verify or authorize documentation. Hence, there might be a delay in release of a new version as the documentation of the previous release is not authorized yet.
-- add more
+Bioinformaticians work in a fairly different environement than biologists. The workflows they create, and the documentation that describe them, are all stored in files on a computer, and sometimes worked on by several person at the same time. So to follow the changes on both workflow and documentation, a more fitting system than the classic Quality Management System (QMS) offered for the biologists must be used. As specified in the ISO-15189, "Software procedures and related documentation needs to be controlled". This can be in any form or type of medium, as long as we achieve a certain set of requirements. In this document, we offer guidelines on how to comply with the norms while having a bioinformatician friendly QMS.
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: I would start with the 'why'. We do not have to give a lot of background. We need to make a point, and continue with what needs to be done. 
Where we write code is not a topic of discussion in this section.  It should be "QMS" documentation only.

Comment (tobias-beers) [inline docs/fieldnorm/control_of_documents.md]: This part can be merged into a general introduction for the whole fieldnorm

File: docs/fieldnorm/control_of_documents.md [hunk 3]
```diff
@@ -1,25 +1,65 @@
...
-6. documents are protected from unauthorized changes and any deletion or removal;
-7. documents are protected from unauthorized access;
-8. the unintended use of obsolete documents is prevented, and suitable identification is applied to them if they are retained for any purpose;
-9. at least one paper or electronic copy of each obsolete controlled document is retained for a specified time period or in accordance with applicable specified requirements.
+To allow for a "Bioinformatician friendly" QMS, the following aspects need to be considered and documented :
 
-Examples are:
+1. Documents are uniquely identified
+2. Documents are approved for adequacy before issue by authorized personnel who have the expertise and competence to determine adequacy
+3. Documents are periodically reviewed and updated as necessary
```
Comment (tobias-beers) [inline docs/fieldnorm/control_of_documents.md:19]: Should we have guidelines on how often this (minimally) happens?

File: docs/fieldnorm/control_of_documents.md [hunk 4]
```diff
@@ -1,25 +1,65 @@
...
+4. Relevant versions of applicable documents are available at points of use and, where necessary, their distribution is controlled
+5. Changes and the current revision status of documents are identified
+6. Documents are protected from unauthorized changes and any deletion or removal
+7. Documents are protected from unauthorized access
```
Comment (tobias-beers) [inline docs/fieldnorm/control_of_documents.md:23]: 6 + 7 feel a bit double to me, as protection from unauthorized acces will imply protection from unauthorized changes, right?

File: docs/fieldnorm/control_of_documents.md [hunk 5]
```diff
@@ -1,25 +1,65 @@
...
-9. at least one paper or electronic copy of each obsolete controlled document is retained for a specified time period or in accordance with applicable specified requirements.
+To allow for a "Bioinformatician friendly" QMS, the following aspects need to be considered and documented :
 
-Examples are:
+1. Documents are uniquely identified
+2. Documents are approved for adequacy before issue by authorized personnel who have the expertise and competence to determine adequacy
+3. Documents are periodically reviewed and updated as necessary
+4. Relevant versions of applicable documents are available at points of use and, where necessary, their distribution is controlled
+5. Changes and the current revision status of documents are identified
+6. Documents are protected from unauthorized changes and any deletion or removal
```
Comment (tobias-beers) [inline docs/fieldnorm/control_of_documents.md:22]: "*and any* deletion or removal": it sounds like *any* is in contrast to *unauthorized*, I read this now as if both unauthorized and authorized deletion are prevented. But there are situations where authorized deletion of a document is okay. I would change it to "...protected from unauthorized changes and deletion."

File: docs/fieldnorm/control_of_documents.md (hunk 4 above)
Comment (tobias-beers) [reply docs/fieldnorm/control_of_documents.md:23]: or is 7 about reading-access? Then they are both important, but then maybe explicitly add that

File: docs/fieldnorm/control_of_documents.md (hunk 1 above)
Comment (tobias-beers) [inline docs/fieldnorm/control_of_documents.md]: I would nuance the first sentence, or mention that it is an example. "*A lot of* bioinformaticians use git."

Comment (tobias-beers) [reply docs/fieldnorm/control_of_documents.md]: After mentioning the Standard Operating Procdures, add a "(SOP)" to define the abbreviation that you use later

File: docs/fieldnorm/control_of_documents.md (hunk 3 above)
Comment (TLeenders) [reply docs/fieldnorm/control_of_documents.md:19]: I would leave this open. As long as laboratories can reason why they choose a specific time period, this should be fine.

File: docs/fieldnorm/control_of_documents.md (hunk 5 above)
Comment (TLeenders) [reply docs/fieldnorm/control_of_documents.md:22]: These points are taken straight from ISO15189:2022. So I think we can leave it in like this

File: docs/fieldnorm/control_of_documents.md (hunk 1 above)
Comment (TLeenders) [inline docs/fieldnorm/control_of_documents.md]: "..often linked to git." maybe add: "often linked to git, **code repositories and task management systems**."

File: docs/fieldnorm/control_of_documents.md [hunk 6]
```diff
@@ -1,25 +1,65 @@
...
+- 10 - Data Access
+    - 11 - GenomeScan -> 11_genomescan.md
+    - 12 - InternalSequencers -> 12_internalsequencers.md
+- 20 - Data Processing
+    - 21 - WES -> 21_wes_flow.md
+    - 22 - RNAseq -> 22_rnaseq_analysis.md
+
+2. The repository is set such only restricted people who manages the documentation approves new entries to be added into the main document (dev -> main). And rules are set so each new feature or entry is approved by the competent person before being accepted (feature -> dev)
+
+3. The repository is set such as anyone who see something out to date can work on updating the document and make a request to the documentation managers so they can accept the update.  Besides, at the footer of each page, you can see the date of last modification.
```
Comment (TLeenders) [inline docs/fieldnorm/control_of_documents.md]: I'm not sure whether this suffices as having "periodic review".

File: docs/fieldnorm/control_of_documents.md [hunk 7]
```diff
@@ -1,25 +1,65 @@
...
+## Why do Bioinformaticians need a different QMS
+
+Bioinformaticians create workflows collaboratively on a computer within a git environment. Git is a distributed version control system that tracks changes in any set of computer files. This can be used to track the changes of the code itself, but also in its documentation, which can describe the Standard Operating Procedures to use it. Git comes in with a set of features allowing to track issues and their resolution, new release of a workflow and its newest features. As the workflow often get upgraded and updated, their documentation, as well, is susceptible to change. Which is why many Bioinforaticians have turned their back from a standard QMS to have their own documentation system, often linked to git. Such system also offer possibilities to copy-paste code blocks while performing the SOP for a given flow, making it so much easier and less prone to errors. Examples of those documentations systems are:
 
 1. Github pages
-2. Gitlab
-3. Confluence
\ No newline at end of file
+2. Gitlab pages
+3. Confluence
```
Comment (TLeenders) [inline docs/fieldnorm/control_of_documents.md:33]: Looking mainly at ourselves, I find it really hard to comply with points 2,3 and 6. Does anyone know if there is a good free tool for Confluence that allows for managing document authorization in Confluence?

**Review by TLeenders — COMMENTED** [review]
I find it hard to determine the line where documentation should fall under 8.3.2 for bioinf and where not. The norm specifies that "All documents associated with the examination process shall be subject to document control (see 8.3)." and "The laboratory shall control the documents (internal and external) that relate to the fulfilment of this document (ISO15189:2022)." But does this also include documents that are heavily focused on the development side of things? as these are somewhat related to the examination processes, but also not really. 

Are these struggles only specific to us? or do other bioinformatic groups struggle with this as well?

File: docs/fieldnorm/control_of_documents.md (hunk 4 above)
Comment (RxLoutre) [reply docs/fieldnorm/control_of_documents.md:23]: I think indeed it is more to say that external from the organization cannot just reach our documentation. But I am not sure, as @TLeenders said, this is just straight from the ISO, not sure about how to interpret this specific part :)

File: docs/fieldnorm/control_of_documents.md (hunk 6 above)
Comment (RxLoutre) [reply docs/fieldnorm/control_of_documents.md]: I think it definitely does not 😛 . But I could not find a system to set reminders to update a certain page, in gitlab at least. This part is openn to suggestion !

File: docs/fieldnorm/control_of_documents.md (hunk 3 above)
Comment (hwmvandeutekom) [reply docs/fieldnorm/control_of_documents.md:19]: this should be in line with the QMS from the laboratory

File: docs/fieldnorm/control_of_documents.md [hunk 8]
```diff
@@ -1,25 +1,65 @@
...
+
+Many of those tools, if used in a specific way, can allow Bioinformaticians to work in a comfortable environement, while complying to the ISO norms requirements. They are of course multiple ways to implement such a system, and as long as it is documented how your system complies to the ISO requirements, all is well. If you are looking for some inspirations, in the next paragraph we will describe an example on how to use one this tool to follow the ISO requirements for both controlling the workflows/software you develop, and their associated documentation/SOP.
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: I would refer to ISO-15189 instead of "ISO norms" 
"comply to ISO-15189". (you can leave out the requirements which is a bit redundant in this sentence)

Comment (hwmvandeutekom) [reply docs/fieldnorm/control_of_documents.md]: They are of course multiple ways to implement such a system, and as long as it is documented how your system complies to the ISO requirements, all is well.  --> There are multiple ways to implement an alternative system. Make sure you document how your system complies to the ISO-15189.

File: docs/fieldnorm/control_of_documents.md [hunk 9]
```diff
@@ -1,25 +1,65 @@
...
+
+## Example case : The usage of GitLab
+
+### Central gathering of all SOP on Gitlab pages
+_I wanted to add images to illustrates but I did not found out how_ :(
+
+Using gitlab pages and mkdocs, you can easily create a beautiful hub to store all your SOP. If using the following guidelines, then you will comply with each of the ISO requirements mentionned above :
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: If using the following guidelines, then you will comply with each of the ISO requirements mentionned above <- never state this... as then people can point to us saying "they told us this is good enough".
It's not a report that people will read, it's a guideline/fieldnorm. So we should try to avoid making texts long and pretty.

Comment (hwmvandeutekom) [reply docs/fieldnorm/control_of_documents.md]: "all your SOPs"

File: docs/fieldnorm/control_of_documents.md [hunk 10]
```diff
@@ -1,25 +1,65 @@
...
+Many of those tools, if used in a specific way, can allow Bioinformaticians to work in a comfortable environement, while complying to the ISO norms requirements. They are of course multiple ways to implement such a system, and as long as it is documented how your system complies to the ISO requirements, all is well. If you are looking for some inspirations, in the next paragraph we will describe an example on how to use one this tool to follow the ISO requirements for both controlling the workflows/software you develop, and their associated documentation/SOP.
+
+## Example case : The usage of GitLab
+
+### Central gathering of all SOP on Gitlab pages
+_I wanted to add images to illustrates but I did not found out how_ :(
+
+Using gitlab pages and mkdocs, you can easily create a beautiful hub to store all your SOP. If using the following guidelines, then you will comply with each of the ISO requirements mentionned above : 
+
+1. Each SOP will become a markdown file. It's name will include a number for easy and unique identification of the document. The numbering can also be used to classify each SOP into categories. Example :
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: will become --> becomes

File: docs/fieldnorm/control_of_documents.md [hunk 11]
```diff
@@ -1,25 +1,65 @@
...
+- 10 - Data Access
+    - 11 - GenomeScan -> 11_genomescan.md
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: we don't want company names, or other identifiable/unique names

File: docs/fieldnorm/control_of_documents.md [hunk 12]
```diff
@@ -1,25 +1,65 @@
...
+    - 12 - InternalSequencers -> 12_internalsequencers.md
+- 20 - Data Processing
+    - 21 - WES -> 21_wes_flow.md
+    - 22 - RNAseq -> 22_rnaseq_analysis.md
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: for consistancy make it rnaseq_flow

File: docs/fieldnorm/control_of_documents.md (hunk 6 above)
Comment (hwmvandeutekom) [reply docs/fieldnorm/control_of_documents.md]: I've seen a lab where they have a list with pages and dates, which is checked on a monthly basis to see which page needs revision. As long as you do that and you can show that it works, that's fine I guess. 
(so make sure nothing exceeds the periodic review when the RvA drops by ;) )

File: docs/fieldnorm/control_of_documents.md [hunk 13]
```diff
@@ -1,25 +1,65 @@
...
+
+2. The repository is set such only restricted people who manages the documentation approves new entries to be added into the main document (dev -> main). And rules are set so each new feature or entry is approved by the competent person before being accepted (feature -> dev)
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: I would rephrase; (something like)
"new entries to the SOP will be approved by specific team members" 
"after approval, other team members can accept the changes" 
--> and then refer to a list or something to show who is allowed to do what

File: docs/fieldnorm/control_of_documents.md (hunk 6 above)
Comment (hwmvandeutekom) [reply docs/fieldnorm/control_of_documents.md]: the last note is indeed not a way to comply to the ISO15189

maybe put this info at another point

File: docs/fieldnorm/control_of_documents.md [hunk 14]
```diff
@@ -1,25 +1,65 @@
...
+
+3. The repository is set such as anyone who see something out to date can work on updating the document and make a request to the documentation managers so they can accept the update.  Besides, at the footer of each page, you can see the date of last modification.
+
+4. The documentation is always available on gitlab by members of the Bioinformatic team. They can access only the production release to use as documentation material.
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: rephrase the last sentence: Only the SOP that has been approved and accepted is shown as the current version.

File: docs/fieldnorm/control_of_documents.md [hunk 15]
```diff
@@ -1,25 +1,65 @@
...
+
+5. On the gitlab repository of the documentation, it would be a convention to create a tag and a release for every production version released. Going to this tag directory, you can see what changed for each released and what has been changed in which document. Everything can be traced back using commits.
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: it would be a convention to  --> too vague
should be something like "tags are created for ..."

Comment (hwmvandeutekom) [reply docs/fieldnorm/control_of_documents.md]: can be tracked --> Using commits, all changes are tracked.

File: docs/fieldnorm/control_of_documents.md [hunk 16]
```diff
@@ -1,25 +1,65 @@
...
+
+6. No one else that the documentation managers can remove or modify the production documentation. Each modification (which includes deletion) must be reviewed twice before being actually effective in production. So there is a protectiob against accidental removals.
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: Current versions of Procedures cannot be deleted. Each modification, including deletion, will be reviewed twice before becoming effective.

File: docs/fieldnorm/control_of_documents.md [hunk 17]
```diff
@@ -1,25 +1,65 @@
...
+
+7. Only members of the gitlab team can access the documentation.
+
+8. For each tag, you can access the version of the documentation at a certain timee. So all is archived by default in an organized way.
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: This probably also works without tracks. As full history is present in these systems..

File: docs/fieldnorm/control_of_documents.md [hunk 18]
```diff
@@ -1,25 +1,65 @@
...
+
+
+### Control of workflows
+
+_Same here... I made beautifull images to explain and I can't include them here_ :(
```
Comment (hwmvandeutekom) [inline docs/fieldnorm/control_of_documents.md]: not sure if we need workflows in the documentation part of the ISO. Or maybe I forget something :)

File: docs/fieldnorm/control_of_documents.md (hunk 2 above)
Comment (geertgeeven) [reply docs/fieldnorm/control_of_documents.md]: I agree. With this I think we make a compelling argument for why we need a field norm for bioinformatics and we can make this argument in the general introduction

File: docs/fieldnorm/control_of_documents.md (hunk 3 above)
Comment (pneerincx) [reply docs/fieldnorm/control_of_documents.md:19]: Maybe we can add that it needs to be updated/checked at least whenever a new version/release of software/code/pipelines is pushed to production. Or maybe even better that the documents are checked as part of a validation/verification and hence before it is used for production?...

File: docs/fieldnorm/control_of_documents.md [hunk 19]
```diff
@@ -1,25 +1,65 @@
 # Control of documents
 
-Bioinformaticians work in a git environment. Git is a distributed version control system that tracks changes in any set of computer files, usually used for coordination work among programmers collaboratively developing source code during software development. Software procedures and related documentation needs to be controlled, and according to ISO-15189, this can be in any form or type of medium. For software development, there are several reasons to not include all the software documentation within the medical laboratories chosen quality management system.
+## Introduction 
 
-- Software development is done in a distributed version control system. These systems are used for coordination work among programmers. Often these systems also allow for documentation of procedures. With the following advantages a. Automatically link to i.e. code, issues, releases, and release notes. b. Possibility to include code blocks that allow for copy-paste into the console/terminal. It’s fast and readable for bioinformaticians. For example, code blocks can be used for general procedures to start a pipeline, or for trouble shooting.
```
Comment (RxLoutre) [inline docs/fieldnorm/control_of_documents.md:5]: Add this back -> more arguments to explain why we need a different a different QMS

File: docs/fieldnorm/control_of_documents.md [hunk 20]
```diff
@@ -1,25 +1,65 @@
...
-- Software development, and hence, corresponding documentation is prone to frequent updates. With the QMS of the laboratory, there are often waiting steps for different members of the laboratory to verify or authorize documentation. Hence, there might be a delay in release of a new version as the documentation of the previous release is not authorized yet.
```
Comment (RxLoutre) [inline docs/fieldnorm/control_of_documents.md:6]: Also add this back for more argumentations

File: docs/fieldnorm/control_of_documents.md (hunk 1 above)
Comment (TLeenders) [reply docs/fieldnorm/control_of_documents.md]: also add the reason that we want to be able to use specific formatting like codeblocks which is not possible in (most) QMS systems

File: docs/fieldnorm/control_of_documents.md (hunk 19 above)
Comment (RxLoutre) [reply docs/fieldnorm/control_of_documents.md:5]: Insist code blocks have no errors on this kind of format compared to copy pasted from a word document

Comment (RxLoutre) [reply docs/fieldnorm/control_of_documents.md:5]: Referring to issues etc in our documentation.
Mention that when doing a software update, we must have the corresponding SOP in our docs be updated as well. It might be possible to do automatically.

File: docs/fieldnorm/control_of_documents.md (hunk 4 above)
Comment (pneerincx) [reply docs/fieldnorm/control_of_documents.md:23]: I think 7 is indeed about read access. In my opinion that is only relevant when there is _sensitive_ data in your SOP. E.g. if the SOP list the admin/root password required to fix a broken server, then obviously that password should not be readable by unauthorized persons. In general though I would advocate _open access_ for SOPs just as much as for scientific papers and just like _open source_ for software.

File: docs/fieldnorm/control_of_documents.md (hunk 1 above)
Comment (pneerincx) [inline docs/fieldnorm/control_of_documents.md]:
```suggestion
Bioinformaticians collaboratively create software, which is maintained in distributed version control systems like for example _Git_ as part of best practices to ensure high quality code. A distributed version control system can be used to track not only the changes in the code itself, but also changes in its documentation. _Git_ comes with a set of features allowing it to track issues and their resolution, new releases of a workflow and its newest features. As workflows often get upgraded and updated, their documentation, as well, is susceptible to change. This documentation can describe amongst other the Standard Operating Procedures to use the software as well as the results from validation experiments used to prove that a certain feature works or that a bug was fixed. Hence a QMS suitable for Bioinformaticians must support
 - Tight integration between code, SOPs and validation/verification reports.
 - Offer possibilities to document code blocks required for performing the SOP for a given workfow. Bioinformaticians must be able to copy-paste the code without the formatting/layout getting messed up to ensure error-free execution of the code.
Examples of those documentations systems are:
```

File: docs/fieldnorm/control_of_documents.md (hunk 4 above)
Comment (TLeenders) [reply docs/fieldnorm/control_of_documents.md:23]: I don't think we have to stress on this point to much. In general this means that there should be a process for granting people access to your documentation system. So as long as your git repo's with all documentation aren't open source, you should be fine.

**Comment by rernst** [comment]
Closing this one, see new PR #12
